src/kboot.c

```diff
@@ -349,6 +349,52 @@ static int dt_set_serial_number(void)
     return 0;
 }
 
+static int dt_set_smbios(void)
+{
+    int fdt_root = fdt_path_offset(dt, "/");
+    int len, node, smbios_node;
+
+    if (fdt_root < 0)
+        bail("FDT: could not open a handle to FDT root.\n");
+
+    node = fdt_add_subnode(dt, fdt_root, "smbios");
+    if (node < 0)
+        bail("FDT: failed to add node 'smbios' to  '/'\n");
+    fdt_setprop_string(dt, node, "compatible", "u-boot,sysinfo-smbios");
+
+    smbios_node = fdt_add_subnode(dt, node, "smbios");
+    if (smbios_node < 0)
+        bail("FDT: failed to add node 'smbios' to  '/smbios'\n");
+
+    node = fdt_add_subnode(dt, smbios_node, "system");
+    if (node < 0)
+        bail("FDT: failed to add node 'system' to  '/smbios'\n");
+    fdt_setprop_string(dt, node, "manufacturer", "Apple");
+    const char *model = fdt_getprop(dt, fdt_root, "model", &len);
+    if (!model)
+        bail("FDT: failed to get model property\n");
+    if (len < 6)
+        fdt_setprop_string(dt, node, "product", "Unknown");
+    else
+        fdt_setprop_string(dt, node, "product", model + 6);
+
+    node = fdt_add_subnode(dt, smbios_node, "baseboard");
+    if (node < 0)
+        bail("FDT: failed to add node 'baseboard' to  '/smbios'\n");
+    fdt_setprop_string(dt, node, "manufacturer", "Apple");
+    const char *target_type = adt_getprop(adt, 0, "target-type", NULL);
+    if (!target_type)
+        bail("ADT: failed to get target-type property\n");
+    fdt_setprop_string(dt, node, "product", target_type);
+
+    node = fdt_add_subnode(dt, smbios_node, "chassis");
+    if (node < 0)
+        bail("FDT: failed to add node 'chassis' to  '/smbios'\n");
+    fdt_setprop_string(dt, node, "manufacturer", "Apple");
```

Comment:
Please set "chassis-type" as well. Best information to derive it from is "/product/product-name". If it starts with "MacBook" use `laptop` (`0x09`), if it starts with "iMac" use `All in One` (`0x0d`) and otherwise `desktop` (`0x03`). Let's pretend the Mac Pro does not exists.

Reply:
That's something which is hopefully simple to [change](https://github.com/devicetree-org/dt-schema/pull/162)

Reply:
Since "all-in-one" was accepted, I've added code to set the smbios enclosure type based on the "chassis-type" property in the device tree. Which means that property needs to be added to our device trees before this will show up as something else than "unknown" in smbios.

Also note that further u-boot changes are necessary to make this show up correctly anyway as the chassis type is currently only set if CONFIG_GENERATE_SMBIOS_TABLE_VERBOSE=y.

Reply:
BTW, I did add code for "server" just in case that's what folks think is appropripriate for the Mac Pro.